src/test/java/healthy/lifestyle/backend/admin/users/service/UserAdminServiceTest.java

```diff
@@ -0,0 +1,157 @@
+package healthy.lifestyle.backend.admin.users.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+import healthy.lifestyle.backend.admin.users.repository.UserAdminRepository;
+import healthy.lifestyle.backend.exception.ApiException;
+import healthy.lifestyle.backend.exception.ErrorMessage;
+import healthy.lifestyle.backend.users.dto.UserResponseDto;
+import healthy.lifestyle.backend.users.model.Country;
+import healthy.lifestyle.backend.users.model.Role;
+import healthy.lifestyle.backend.users.model.User;
+import healthy.lifestyle.backend.users.repository.CountryRepository;
+import healthy.lifestyle.backend.users.repository.RoleRepository;
+import healthy.lifestyle.backend.util.TestUtil;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+
+@ExtendWith(MockitoExtension.class)
+class UserAdminServiceTest {
```

Comment:
Write one positive test case with random list of users dto, and one negative test case when repository not found users.